pkg/agent/controller/bgp/controller.go

```diff
@@ -959,3 +959,18 @@ func (c *Controller) GetBGPPolicyInfo() (string, string, int32, int32) {
 	}
 	return name, routerID, localASN, listenPort
 }
+
+// GetBGPPeerStatus returns current status of all BGP Peers of effective BGP Policy applied on the Node.
+func (c *Controller) GetBGPPeerStatus() ([]bgp.PeerStatus, error) {
+	c.bgpPolicyStateMutex.RLock()
+	defer c.bgpPolicyStateMutex.RUnlock()
+
+	if c.bgpPolicyState == nil {
+		return nil, fmt.Errorf("failed to get bgp peers: there is no effective bgp policy applied to the Node")
+	}
+	peers, err := c.bgpPolicyState.bgpServer.GetPeers(context.TODO())
```

Comment:
holding a read lock while calling this function may be bit overkill. In the case of gobgp at least, all these operations are asynchronous (so there can be some delay, hence the context parameter) and all the server functions can be called concurrently as far as I know.

so the following would probably be correct:
```go
getBgpServer := func() bgp.Interface {
        c.bgpPolicyStateMutex.RLock()
	defer c.bgpPolicyStateMutex.RUnlock()
        if c.bgpPolicyState == nil {
		return nil
	}
	return c.bgpPolicyState.bgpServer
}

bgpServer := getBgpServer()
if bgpServer == nil {
	// return error
}
peers, err := bgpServer.GetPeers(ctx)
```

This way we don't hold the read lock for a potentially expensive async operation.
That means that we have to guarantee that all `bgp.Interface` implementations (at the moment we only have gobgp) support concurrent function calls. Probably worth double checking that this assumption holds for gobgp, but it seems to me that all API calls get serialized to a channel.